sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -325,8 +327,10 @@ case class FileSourceScanExec(
   override lazy val metrics =
     Map("numOutputRows" -> SQLMetrics.createMetric(sparkContext, "number of output rows"),
       "numFiles" -> SQLMetrics.createMetric(sparkContext, "number of files"),
-      "metadataTime" -> SQLMetrics.createMetric(sparkContext, "metadata time"),
-      "scanTime" -> SQLMetrics.createTimingMetric(sparkContext, "scan time"))
+      "scanTime" -> SQLMetrics.createTimingMetric(sparkContext, "scan time"),
+      "metadataTime" -> SQLMetrics.createMetric(sparkContext, "metadata time (ms)"),
+      "fileListingStart" -> SQLMetrics.createTimestampMetric(sparkContext, "file listing start"),
```

Comment:
Here's a problem existing, for the metrics like metadataTime and start/end timestamp, they are driver-only, which only create, update and display controlled on driver-side. The current usage has little waste on sending metrics to the executor and aggregates back. I'll address this in a separate JIRA and PR, this is a common work for both SPARK-26222/SPARK-26223.